src/main/java/io/wisetime/connector/inprotech/InprotechConnector.java

```
@@ -167,13 +174,19 @@ public PostResult postTime(Request request, TimeGroup userPostedTimeUtc) {
     // The important information is the time spent, which we set as the (possibly edited) total time.
     final LocalDateTime endTime = activityStartTime.get().plus(workedTime, ChronoUnit.SECONDS);
 
+    final String narrativeMsg = narrativeTemplateFormatter.format(zonedTimeGroup);
+    final String notesMsg = internalNotesTemplateFormatter.format(zonedTimeGroup);
+    if (notesMsg.length() > NOTES_MAX_LENGTH) {
+      return PostResult.PERMANENT_FAILURE.withMessage("Maximum length for notes exceeded.");
```

Fail early if generated notes exceeded max length